File: app/src/main/java/ru/nikshlykov/englishwordsapp/db/ModeDao.java
package ru.nikshlykov.englishwordsapp.db;

import androidx.room.Dao;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface ModeDao {

    @Update
    int update(List<Mode> modes);

    // Получаем все режимы.
    @Query("SELECT * FROM Modes")
    List<Mode> getAllModes();

    @Query("SELECT * FROM Modes WHERE IsSelected = 1")
    Mode[] getSelectedModes();
}

File: app/src/main/java/ru/nikshlykov/englishwordsapp/db/daos/ModeDao.kt
package ru.nikshlykov.englishwordsapp.db.daos

import androidx.room.Dao
import androidx.room.Query
import androidx.room.Update
import ru.nikshlykov.englishwordsapp.db.models.Mode

@Dao
interface ModeDao {

  @Update
  suspend fun update(modes: List<Mode>): Int

  @Query("SELECT * FROM Modes WHERE isSelected = 1")
  suspend fun getSelectedModes(): List<Mode>

  @Query("SELECT * FROM Modes")
  suspend fun getAllModes(): List<Mode>
}